File: Python/Hanoi/hanoi.py
def hanoi(n, froms, to, via):
    if n > 1:
        hanoi(n - 1, froms, via, to)
        print(froms + " -> " + to)
        hanoi(n - 1, via, to, froms)
    else:
        print(froms + " -> " + to)

n = int(input())
hanoi(n, "a", "b", "c")

File: JavaScript/Hanoi/hanoi.js
function hanoi(n, from, to, via) {
    if (n > 1) {
        hanoi(n - 1, from, via, to);
        console.log(from + " -> " + to);
        hanoi(n - 1, via, to, from);
    } else {
        console.log(from + " -> " + to);
    }
}

process.stdin.on('data', function (line) {
    hanoi(line, "a", "b", "c");
    process.exit(0);
});
